pkg/nsx/services/ippool/ippool.go

```diff
@@ -0,0 +1,252 @@
+package ippool
+
+import (
+	"sync"
+	"time"
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha2"
+	commonctl "github.com/vmware-tanzu/nsx-operator/pkg/controllers/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/util"
+)
+
+var (
+	log                           = logger.Log
+	MarkedForDelete               = true
+	EnforceRevisionCheckParam     = false
+	ResourceTypeIPPool            = common.ResourceTypeIPPool
+	ResourceTypeIPPoolBlockSubnet = common.ResourceTypeIPPoolBlockSubnet
+	NewConverter                  = common.NewConverter
+)
+
+type IPPoolService struct {
+	common.Service
+	ipPoolStore            *IPPoolStore
+	ipPoolBlockSubnetStore *IPPoolBlockSubnetStore
+}
+
+func InitializeIPPool(service common.Service) (*IPPoolService, error) {
+	wg := sync.WaitGroup{}
+	wgDone := make(chan bool)
+	fatalErrors := make(chan error)
+
+	wg.Add(2)
+
+	ipPoolService := &IPPoolService{Service: service}
+	ipPoolService.ipPoolStore = &IPPoolStore{ResourceStore: common.ResourceStore{
+		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{common.TagScopeIPPoolCRUID: indexFunc}),
+		BindingType: model.IpAddressPoolBindingType(),
+	}}
+	ipPoolService.ipPoolBlockSubnetStore = &IPPoolBlockSubnetStore{ResourceStore: common.ResourceStore{
+		Indexer:     cache.NewIndexer(keyFunc, cache.Indexers{common.TagScopeIPPoolCRUID: indexFunc}),
+		BindingType: model.IpAddressPoolBlockSubnetBindingType(),
+	}}
+
+	go ipPoolService.InitializeResourceStore(&wg, fatalErrors, ResourceTypeIPPool, ipPoolService.ipPoolStore)
+	go ipPoolService.InitializeResourceStore(&wg, fatalErrors, ResourceTypeIPPoolBlockSubnet, ipPoolService.ipPoolBlockSubnetStore)
```

Comment:
For NSX VPC Subnet, there will also be IP Pool and Block Subnet created.  I suppose we do not want to search and store them in ipPoolService.ipPoolStore and ipPoolService.ipPoolBlockSubnetStore. 
So we need to search via tags?

Reply:
Updated in https://github.com/vmware-tanzu/nsx-operator/pull/215

Reply:
I do not see ippoolService initialize resource store via tags in #215.

Reply:
Done. Add
		{Scope: String(common.TagScopeIPSubnetOwner), Tag: String(ResourceTypeIPPool)},
You can do it in the same way.